src/ngraph/op/max_pool.cpp

```diff
@@ -492,3 +504,113 @@ void op::v1::MaxPool::generate_adjoints(autodiff::Adjoints& adjoints, const Outp
 
     adjoints.add_delta(operand, backprop);
 }
+
+namespace
+{
+    template <element::Type_t ET>
+    inline bool evaluate(const HostTensorPtr& arg,
+                         const HostTensorPtr& out,
+                         const Shape& out_shape,
+                         const Shape& window_shape,
+                         const Strides& window_movement_strides,
+                         const Shape& padding_below,
+                         const Shape& padding_above)
+    {
+        using T = typename element_type_traits<ET>::value_type;
+        out->set_shape(out_shape);
+        runtime::reference::max_pool<T>(arg->get_data_ptr<ET>(),
+                                        out->get_data_ptr<ET>(),
+                                        arg->get_shape(),
+                                        out_shape,
+                                        window_shape,
+                                        window_movement_strides,
+                                        padding_below,
+                                        padding_above);
+        return true;
+    }
+
+    bool evaluate_maxpool(const HostTensorPtr& arg,
+                          const HostTensorPtr& out,
+                          const Shape& out_shape,
+                          const Shape& kernel,
+                          const Strides& strides,
+                          const Shape& pad_begin,
+                          const Shape& pad_end)
+    {
+        bool rc = true;
+        auto arg_shape = arg->get_shape();
+
+        switch (out->get_element_type())
+        {
+            TYPE_CASE(i8)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+            TYPE_CASE(i16)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+            TYPE_CASE(i32)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+            TYPE_CASE(i64)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+            TYPE_CASE(u8)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+            TYPE_CASE(u16)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+            TYPE_CASE(u32)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+            TYPE_CASE(u64)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+            TYPE_CASE(f32)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+            TYPE_CASE(f64)(arg, out, out_shape, kernel, strides, pad_begin, pad_end);
+            break;
+        default: rc = false; break;
+        }
+        return rc;
+    }
+} // namespace
+
+bool op::v0::MaxPool::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs)
+{
+    auto arg_shape = inputs[0]->get_partial_shape();
+    update_auto_padding(arg_shape);
```

Comment:
Can't do this. Evaluating should not modify the node.

Reply:
will make it stateless.